crates/nu-command/src/conversions/into/duration.rs

```diff
@@ -190,25 +404,110 @@ fn string_to_duration(s: &str, span: Span, value_span: Span) -> Result<i64, Shel
         src_span: value_span,
-        help: Some(
-            "supported units are ns, us/µs, ms, sec, min, hr, day, wk, month, yr, and dec"
-                .to_string(),
-        ),
+        help: Some("supported units are ns, us/µs, ms, sec, min, hr, day, wk".to_string()),
     })
 }
 
-fn action(input: &Value, span: Span) -> Value {
+fn string_to_unit_duration(
+    s: &str,
+    span: Span,
+    value_span: Span,
+) -> Result<(&str, i64), ShellError> {
+    if let Some(Ok(expression)) = parse_unit_value(
+        s.as_bytes(),
+        span,
+        DURATION_UNIT_GROUPS,
+        Type::Duration,
+        |x| x,
+    ) {
+        if let Expr::ValueWithUnit(value, unit) = expression.expr {
+            if let Expr::Int(x) = value.expr {
+                match unit.item {
+                    Unit::Nanosecond => return Ok(("ns", x)),
+                    Unit::Microsecond => return Ok(("µs", x)),
+                    Unit::Millisecond => return Ok(("ms", x)),
+                    Unit::Second => return Ok(("sec", x)),
+                    Unit::Minute => return Ok(("min", x)),
+                    Unit::Hour => return Ok(("hr", x)),
+                    Unit::Day => return Ok(("day", x)),
+                    Unit::Week => return Ok(("wk", x)),
+
+                    _ => return Ok(("ns", 0)),
+                }
+            }
+        }
+    }
+
+    Err(ShellError::CantConvertToDuration {
+        details: s.to_string(),
+        dst_span: span,
+        src_span: value_span,
+        help: Some("supported units are ns, us/µs, ms, sec, min, hr, day, wk".to_string()),
+    })
+}
+
+fn action(
+    input: &Value,
+    convert_to_unit: &Option<Spanned<String>>,
+    float_precision: usize,
+    span: Span,
+) -> Value {
     match input {
         Value::Duration { .. } => input.clone(),
         Value::String {
             val,
             span: value_span,
-        } => match string_to_duration(val, span, *value_span) {
-            Ok(val) => Value::Duration { val, span },
-            Err(error) => Value::Error {
-                error: Box::new(error),
-            },
-        },
+        } => {
+            if let Some(to_unit) = convert_to_unit {
+                if let Ok(dur) = string_to_unit_duration(val, span, *value_span) {
+                    let from_unit = dur.0;
+                    let duration = dur.1;
+                    match convert_str_from_unit_to_unit(
+                        duration,
+                        from_unit,
+                        &to_unit.item,
+                        span,
+                        *value_span,
+                    ) {
+                        Ok(d) => {
+                            let unit = if &to_unit.item == "us" {
+                                "µs"
+                            } else {
+                                &to_unit.item
+                            };
+                            if d.fract() == 0.0 {
+                                Value::String {
```

Comment:
Maybe I'm looking at this part wrong but isn't this part and the part below still returning a Value::String instead of a Value::Duration? If so, can we please just make `into duration` only return Value::Duration?